.github/workflows/push.yml

```diff
@@ -0,0 +1,29 @@
+name: Build
+on:
+  - push
+  - pull_request
+
+
+jobs:
+  debian_x86:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 30
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - name: Run cargo build
+        run: cargo build
+      - name: Run cargo test
+        run: cargo test
+  macos_x86:
+    runs-on: macos-12
+    timeout-minutes: 30
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
```

Comment:
Is this rust-cache necessary? I'd rather not pull in this dependency.

Reply:
@vlad17 Right now, compiling crate is super fast, but It's better to cache anyway.

There are no limitations, except for 10 GB from GitHub. In particular, caches are currently limited to 10 GB in total, and exceeding that limit will cause eviction of older caches.

Anyway, I can drop it if you are not okay with this change.

Reply:
Removed...